File: Example/common/src/main/java/com/example/common/okhttp/OkHttpManager.java
package com.example.common.okhttp;


import okhttp3.Call;
import okhttp3.OkHttpClient;
import okhttp3.Request;

/**
 * @author tengfei
 * date 2019/2/15 4:01 PM
 * email tengfeigo@outlook.com
 * description
 */
public class OkHttpManager {

    public Call asyncCall(String url){
        OkHttpClient okHttpClient = new OkHttpClient();
        Request request = new Request.Builder()
                .get()
                .url(url)
                .build();
        return okHttpClient.newCall(request);
    }
}

File: OkHttpExample/app/src/main/java/com/example/okhttp/OkHttpManager.kt
package com.example.okhttp

import okhttp3.Call
import okhttp3.OkHttpClient
import okhttp3.Request
import okhttp3.Response

/**
 * @author tengfei
 * date 2019/3/4 3:08 PM
 * email tengfeigo@outlook.com
 * description OkHttp管理类，同步和异步方法封装
 */
class OkHttpManager private constructor() {

    private var okHttpClient: OkHttpClient? = null

    init {
        okHttpClient = OkHttpClient()
    }

    companion object {
        val instance = Holder.okHttpManager
    }

    private object Holder {
        val okHttpManager = OkHttpManager()
    }

    fun asyncRequest(url: String): Call {
        val request = Request.Builder()
                .get()
                .url(url)
                .build()
        return okHttpClient!!.newCall(request)
    }

    fun syncRequest(url: String, start: Long, end: Long): Response {
        val request = Request.Builder()
                .get()
                .url(url)
                .addHeader("Range", "bytes=$start-$end")
                .build()
        val call = okHttpClient!!.newCall(request)
        return call.execute()
    }
}